xmtp_mls/src/identity.rs

```diff
@@ -667,7 +667,11 @@ impl Identity {
         let kp_bytes = kp.tls_serialize_detached()?;
         let hash_ref = serialize_key_package_hash_ref(&kp, &provider)?;
         let history_id = conn
-            .store_key_package_history_entry(hash_ref.clone(), pq_pub_key.clone())?
+            .store_key_package_history_entry(
+                hash_ref.clone(),
+                pq_pub_key.clone(),
+                kp.life_time().not_after() as i64,
```

Comment:
marking the KeyPackage to be rotated and deleted based on their life_time. in the future if we change the keypackage life_time this will automatically inherit the value from that.